database/factories/GameAchievementSetFactory.php

```diff
@@ -33,8 +33,8 @@ public function type(string $type): self
         return $this->state(function (array $attributes) use ($type) {
             return [
                 'type' => $type,
-                'title' => $type === AchievementSetType::Core ? null : $this->faker->words(3, true),
-                'order_column' => $type === AchievementSetType::Core ? 0 : $this->faker->numberBetween(1, 10),
+                'title' => $type === AchievementSetType::CORE->value ? null : $this->faker->words(3, true),
+                'order_column' => $type === AchievementSetType::CORE->value ? 0 : $this->faker->numberBetween(1, 10),
```

Comment:
What does the "->value" do for us here?

Reply:
Most enums in RAWeb are not true enums, but rather abstract classes. PHP 8.1 introduced native enums, which have better type safety, are batteries-included with a set of sane instance methods, and are easily serializable.

When studying how to best utilize native enums, I came across this document which I learned a lot of best practices from: https://jeffochoa.me/php-enums-best-practices

`->value` is the built-in property for accessing the scalar value associated with an enum case. This is much safer than using an abstract class style enum, for example:
```php
// returns "Invalid flag", but nothing stops someone from passing in 7.
AchievementFlag::toString(7);
```

If `AchievementFlag` were a native enum, eg:
```php
enum AchievementFlag: int {
    case OfficialCore = 3;
    case Unofficial = 5;
}
```
This would throw an error:
```php
$AchievementFlag::from(7);
```

Another benefit we get is being able to use native enums as types in function args. Abstract class enums do not support this. eg:
```php
 // invalid!
function myFunction(AchievementFlag $flag) { ... }

 // valid.
function myFunction(AchievementSetType $type) { ... }
```

Ultimately, I believe native enums give us the same benefits that we get from abstract classes, but make the code much safer in the long run.

Reply:
I didn't realize that PHP had strongly typed enums. I'm all for using them.

So the `->value` is necessary to downcast it back to an int? Makes sense.

Reply:
Yep, just skimming over [this article](https://jeffochoa.me/php-enums-best-practices) is a great ~2 minute crash course on all the built-in functionality.